### PR TITLE
infra: pin patch version of openjdk base image

### PR DESCRIPTION
**Pull request by nrmancuso**
*Should* fix failure at https://app.circleci.com/pipelines/github/checkstyle/checkstyle/15829/workflows/6457dceb-0274-4045-9593-b6ae93660de6/jobs/191927, there have been issues reported with newest cimg/openjdk

### CODE REVIEW

**Comment by romani** [comment]
it is a latest - https://github.com/CircleCI-Public/cimg-openjdk/commits/main/11.0

**Comment by romani** [comment]
I do not see this tag at https://hub.docker.com/r/circleci/openjdk/tags?page=1&name=11.0.16
how  can I have more detail on this tag?

**Comment by nrmancuso** [comment]
> I do not see this tag at https://hub.docker.com/r/circleci/openjdk/tags?page=1&name=11.0.16 how can I have more detail on this tag?

https://circleci.com/developer/images/image/cimg/openjdk

File: devops/docker/README.md [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+To build image:
+```bash
+docker build -t <org>/<image-name>:<version> .
+```
+
+To push a new tag:
+```bash
+docker push <org>/<image-name>:<version>
+```
```
Comment (romani) [inline devops/docker/README.md:16]: put here section as example:
https://github.com/checkstyle/contribution/pull/668

to be clear what is org name, image name , and how we do versioning.

**Review by romani — CHANGES_REQUESTED** [review]
items:

File: devops/docker/README.md (hunk 1 above)
Comment (nrmancuso) [reply devops/docker/README.md:23]: good idea, done

**Comment by romani** [comment]
@nrmancuso , do you mean that 11.0.17 might has "defect" so we need to use 11.0.16 ?
did you run it on local ? last time pushing to dockerhub took me few hours.

**Comment by nrmancuso** [comment]
> do you mean that 11.0.17 might has "defect" 

Yes

> did you run it on local ?

No, I can do this now.

File: devops/docker/README.md [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+```bash
+docker login
+```
+
+To build image:
+```bash
+docker build -t <org>/<image-name>:<version> .
+```
+
+Where `version` is <jdk-version>-<idea-version>.
```
Comment (romani) [inline devops/docker/README.md:13]: >idea-version

this is too specific for one docker container ... I thought you tried to make general doc for all.
general versioning is `-` separate versions of packges that is in docker 
example https://hub.docker.com/r/checkstyle/jdk-11-groovy-git-mvn/tags
or https://hub.docker.com/r/checkstyle/idea-docker/tags

**Review by romani — CHANGES_REQUESTED** [review]
items:

File: devops/docker/README.md (hunk 2 above)
Comment (nrmancuso) [reply devops/docker/README.md]: done

**Review by romani — APPROVED** [review]

**Comment by romani** [comment]
pushed: https://hub.docker.com/r/checkstyle/idea-docker/tags

```
docker build -t checkstyle/idea-docker:jdk11.0.16-idea2022.2.2 .
docker login
docker push checkstyle/idea-docker:jdk11.0.16-idea2022.2.2

```

**Comment by rnveach** [comment]
@romani  @nrmancuso CI is failing in this repo.

````
CMD=./.ci/travis.sh markdownlint

devops/docker/README.md:1: MD002 First header should be a top level header

devops/docker/README.md:4: MD031 Fenced code blocks should be surrounded by blank lines

devops/docker/README.md:9: MD031 Fenced code blocks should be surrounded by blank lines

devops/docker/README.md:16: MD031 Fenced code blocks should be surrounded by blank lines

devops/docker/README.md:21: MD031 Fenced code blocks should be surrounded by blank lines

devops/docker/README.md:26: MD031 Fenced code blocks should be surrounded by blank lines

devops/docker/README.md:33: MD031 Fenced code blocks should be surrounded by blank lines

devops/docker/README.md:38: MD031 Fenced code blocks should be surrounded by blank lines

devops/docker/README.md:13: MD033 Inline HTML

devops/docker/README.md:13: MD033 Inline HTML

devops/docker/README.md:1: MD041 First line in file should be a top level header

Further documentation is available for these failures:

 - MD002: https://github.com/markdownlint/markdownlint/blob/master/docs/RULES.md#md002---first-header-should-be-a-top-level-header

 - MD031: https://github.com/markdownlint/markdownlint/blob/master/docs/RULES.md#md031---fenced-code-blocks-should-be-surrounded-by-blank-lines

 - MD033: https://github.com/markdownlint/markdownlint/blob/master/docs/RULES.md#md033---inline-html

 - MD041: https://github.com/markdownlint/markdownlint/blob/master/docs/RULES.md#md041---first-line-in-file-should-be-a-top-level-header

The command "./.ci/travis.sh run-command "$CMD"" exited with 1.
````

**Comment by rnveach** [comment]
I will work on master failure.

**Pull request merged**